parsec/core/mountpoint/winfsp_runner.py

```diff
@@ -154,25 +135,40 @@ async def winfsp_mountpoint_runner(
         # security_timeout=10000,
     )
     try:
-        event_bus.send("mountpoint.starting", mountpoint=mountpoint_path)
+        # Prepare event information
+        event_kwargs = {
+            "mountpoint": mountpoint_path,
+            "workspace_id": workspace_fs.workspace_id,
+            "timestamp": getattr(workspace_fs, "timestamp", None),
+        }
+        event_bus.send("mountpoint.starting", **event_kwargs)
+
+        # Manage drive icon
+        drive_letter, *_ = mountpoint_path.drive
+        with parsec_drive_icon_context(drive_letter):
+
+            # Run fs start in a thread
+            await trio.to_thread.run_sync(fs.start)
+            await _wait_for_winfsp_ready(mountpoint_path)
+
+            # Notify the manager that the mountpoint is ready
+            event_bus.send("mountpoint.started", **event_kwargs)
+            task_status.started(mountpoint_path)
 
-        # Run fs start in a thread, as a cancellable operation
-        # This is because fs.start() might get stuck for while in case of an IRP timeout
-        await trio.to_thread.run_sync(fs.start, cancellable=True)
+            # Start recording `sharing.updated` events
+            with event_bus.waiter_on("sharing.updated") as waiter:
```

Comment:
This doesn't seems to be restricted to winfspy, wouldn't it be better to put it in the generic manager along with the event handler dealing with workspace creation/sharing automount ?

Reply:
It's kind of restricted to winfsp as it is relasted to the fact that we can't change the `readonly` status of the drive at runtime.

Good point though, this reminds me that I still have to investigate how read only mountpoints work on linux.

Reply:
Apparently there is no equivalent to `read_only_volume` on fuse.